File: app/src/main/java/com/simcom/ecashier/model/room/GroupDao.java
package com.simcom.ecashier.model.room;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface GroupDao {
    @Insert
    void insert(Group group);

    @Update
    void update(Group group);

    @Delete
    void delete(Group group);

    @Query("SELECT * FROM GROUP_TABLE ORDER BY name")
    LiveData<List<Group>> getAllGroups();
}

File: app/src/main/java/com/simcom/ecashier/model/room/GroupDao.kt
package com.simcom.ecashier.model.room

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface GroupDao {
    @Insert
    suspend fun insert(group: Group)

    @Update
    suspend fun update(group: Group)

    @Delete
    suspend fun delete(group: Group)

    @Query("SELECT * FROM GROUP_TABLE ORDER BY name")
    fun allGroups(): LiveData<List<Group>>
}